modules/caddyhttp/replacer.go

```diff
@@ -143,6 +143,10 @@ func addHTTPVarsToReplacer(repl *caddy.Replacer, req *http.Request, w http.Respo
 			case "http.request.uri.path.dir":
 				dir, _ := path.Split(req.URL.Path)
 				return dir, true
+			case "http.request.uri.path.file.base":
+				return strings.TrimSuffix(path.Base(req.URL.Path), path.Ext(req.URL.Path)), true
+			case "http.request.uri.path.file.ext":
+				return path.Ext(req.URL.Path), true
```

Comment:
Btw, with an input of `foo.tar.gz`, this will return `.gz` only. Should we write our own logic to repeat `.Ext()` to pop extensions off the end until none remain, and return that? :thinking:

Reply:
Well, there are 2 extensions. Imagine a path of `index.012abcdf.html` -- there's only one file extension and the part in the middle is just a checksum. I'm not sure how to handle that better. It's arbitrary either way, and more commonly there is only 1 ext.